examples/with-cloudinary/components/SharedModal.tsx

```diff
@@ -0,0 +1,210 @@
+import {
+  ArrowDownTrayIcon,
+  ArrowTopRightOnSquareIcon,
+  ArrowUturnLeftIcon,
+  ChevronLeftIcon,
+  ChevronRightIcon,
+  XMarkIcon,
+} from '@heroicons/react/24/outline'
+import { AnimatePresence, motion, MotionConfig } from 'framer-motion'
+import Image from 'next/image'
+import { useState } from 'react'
+import { useSwipeable } from 'react-swipeable'
+import { variants } from '../utils/animationVariants'
+import downloadPhoto from '../utils/downloadPhoto'
+import { range } from '../utils/range'
+import type { ImageProps, SharedModalProps } from '../utils/types'
+import Twitter from './Icons/Twitter'
+
+export default function SharedModal({
+  index,
+  images,
+  changePhotoId,
+  closeModal,
+  navigation,
+  currentPhoto,
+  direction,
+}: SharedModalProps) {
+  const [loaded, setLoaded] = useState(false)
+
+  let filteredImages = images?.filter((img: ImageProps) =>
+    range(index - 15, index + 15).includes(img.id)
+  )
+
+  const handlers = useSwipeable({
+    onSwipedLeft: () => changePhotoId(index + 1),
+    onSwipedRight: () => changePhotoId(index - 1),
+    trackMouse: true,
+  })
+
+  let currentImage = images ? images[index] : currentPhoto
+
+  return (
+    <MotionConfig
+      transition={{
+        x: { type: 'spring', stiffness: 300, damping: 30 },
+        opacity: { duration: 0.2 },
+      }}
+    >
+      <div
+        className="relative z-50 flex aspect-[3/2] w-full max-w-7xl items-center wide:h-full xl:taller-than-854:h-auto"
+        {...handlers}
+      >
+        {/* Main image */}
+        <div className="w-full overflow-hidden">
+          <div className="relative flex aspect-[3/2] items-center justify-center">
+            <AnimatePresence initial={false} custom={direction}>
+              <motion.div
+                key={index}
+                custom={direction}
+                variants={variants}
+                initial="enter"
+                animate="center"
+                exit="exit"
+                className="absolute"
+              >
+                <Image
+                  src={`https://res.cloudinary.com/${
+                    process.env.NEXT_PUBLIC_CLOUDINARY_CLOUD_NAME
+                  }/image/upload/c_scale,${navigation ? 'w_1280' : 'w_1920'}/${
```

Comment:
This feels off because the 1280/1920 distinction is made twice. Once here in `src` and once for `width`.

I'm still thinking that `with-cloudinary` example should use the `loaderFile` to configure cloudinary and then you just pass `src=${currentImage.public_id}.${currentImage.format}` to each <Image>